src/main/java/com/synopsys/integration/detect/lifecycle/run/step/RapidModeStepRunner.java

```diff
@@ -105,6 +113,24 @@ public void runOnline(BlackDuckRunData blackDuckRunData, NameVersion projectVers
         RapidScanResultSummary summary = operationRunner.logRapidReport(rapidFullResults, mode);
 
         operationRunner.publishRapidResults(jsonFile, summary, mode);
+        operationRunner.publishCodeLocationData(formattedCodeLocations);
+    }
+
+    /**
+     * This method takes a list of URLs for a given scan type and adds them to the parsedUrls structure so 
+     * results can be retrieved from BD after all scans are done. It also stores information for the status.json
+     * file in formattedCodeLocations so scanId and type can be reported.
+     */
+    private void processScanResults(List<HttpUrl> scanResultUrls, List<HttpUrl> parsedUrls,
+            Set<FormattedCodeLocation> formattedCodeLocations, String scanType) {
+        parsedUrls.addAll(scanResultUrls);
+        
+        for (HttpUrl httpUrl : scanResultUrls) {
+            String url = httpUrl.toString();
+            String scanId = url.substring(url.lastIndexOf("/") + 1);
```

Comment:
Can we use `UUID.fromString(path.substring(path.lastIndexOf('/') + 1))` here?
The advantage is that the `fromString` method in UUID class also validates for us if the scanID returned is a valid UUID or not. I think it can be similar to the implementation here:
https://github.com/blackducksoftware/synopsys-detect/blob/bf0e9bf972c14935e59dd584564b7c8c39392836/src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java#L349C13-L352C79
Alternative is validating the scan ID manually again here.
It would also be useful to have a new helper method in OperationRunner that extracts the scanID so it can be reused in the future. I have a similar use case for container scans where I'd use such a helper method.

Reply:
I pushed some changes in this area. It was pretty easy to implement the suggestion of using the fromString call and to add an extraction method to OperationRunner. So I was able to use UUID instead of string in this area as I think in general that is a good idea. The isValidScanId method is buried deep in the Signature Scanner code and I did look at lifting it up into a higher area so we could call it in multiple places. Unfortunately this led to a lot of changes around moving Strings to UUID as unfortunately throughout the code we use Strings. Once the impact started to go into the library code I stopped as I think it was becoming a bit large for this specific MR and I was worrying about the increased risk of regressions.